lib/studies/shrine/app.dart

```diff
@@ -93,6 +129,15 @@ class _ShrineAppState extends State<ShrineApp> with TickerProviderStateMixin {
 
   @override
   Widget build(BuildContext context) {
+    if (_isExpandingControllerCompleted.value !=
```

Comment:
Using a statusListener for this feels a bit clunky. Could there be a cleaner way to do this?

Reply:
I've been thinking about this too... Let me see if I can come up with a better API for this, here and everywhere else tab controller is being used.

Reply:
I refactored this out into its own class here: https://github.com/flutter/gallery/pull/389/files#diff-4576115e6281d08372fae69eebb7d35cff0bc534481112f35d3825370064b6abR32

Let me know what you think or if you have any improvement suggestions!

Reply:
Nice! For clarity, I wonder if `setAnimationController` would be better named as `registerStatusListener` or something similar